Added annoations for GeneSigDB

nsklyar opened the pull request:
Refactoring to support annotations from files.

nsklyar committed:
rolled back wrong changes

nsklyar committed:
beginning of refactoring for genesig

nsklyar committed:
work in progress: refactoring for genesigdb

nsklyar committed:
Merge branch 'ensembl' into genesig

nsklyar committed:
refactoring: AnnotationSourceLoader extracted

nsklyar committed:
refactoring: AnnotationSourceLoader extracted

nsklyar committed:
refactoring

nsklyar committed:
Merge branch 'ensembl' into genesig

nsklyar committed:
work in progress: refactoring for genesigdb

nsklyar committed:
emergency commit, code is not working

nsklyar committed:
added AnnotationSourceManager and Converters

nsklyar committed:
changed hibernate mappings for annotation source. Renamed biomartproperty and arraydesign. Tests are working, but oracle DB not updated

nsklyar committed:
changed hibernate mappings for annotation source. Renamed biomartproperty and arraydesign - fixed update script and Tables.sql

nsklyar committed:
Work on AnnotationSource converters

nsklyar committed:
Work on AnnotationSource converters, added tests

nsklyar committed:
Work in progress: Annotator

nsklyar committed:
work in progress, refactoring Annotator (code is not compilable)

nsklyar committed:
work in progress, refactoring Annotator

nsklyar committed:
Loading works. Need to fix FileBasedConnection

nsklyar committed:
removed test

nsklyar committed:
Merge branch 'develop' into genesig

nsklyar committed:
code clean up

nsklyar committed:
Merge branch 'develop' into genesig

nsklyar committed:
code clean up

nsklyar committed:
Merge branch 'ensembl' into genesig

nsklyar committed:
fix import

nsklyar committed:
Revert: wrongly changed names (0433b90)

nsklyar committed:
Revert: wrongly changed names (0433b90)

nsklyar committed:
Merge branch 'genesig' of github.com:gxa/gxa into genesig

nsklyar committed:
revert optimize import

nsklyar committed:
reverted:
 - optimized imports (though some were really needed)
 - new lines

nsklyar committed:
Ticket #3329. Load GeneSigDB annotation data before starting annotation

nsklyar committed:
Merge branch 'develop' into genesig

nsklyar committed:
removed commented out code

olgamelnichuk commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/loader/AnnotationSourceConnection.java:
```diff
@@ -0,0 +1,18 @@
+ * Date: 26/10/2011
+ */
+public abstract class AnnotationSourceConnection <T extends AnnotationSource> {
+
```
why not to make it as Interface?

nsklyar replied:
changed

olgamelnichuk commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/dao/AnnotationSourceDAO.java:
```diff
@@ -51,13 +50,20 @@ public AnnotationSourceDAO(SessionFactory sessionFactory, JdbcTemplate atlasJdbc
+        annSrc.setLoadDate(new Date());
+        //ToDo: Find better solution.This method call is needed to initialize "name" field, if it was not called before name=null.
+        // NS didn't manage to find how to force hibernate to access the field via getter.
+        annSrc.getName();
```
why do we need to save the name into DB if we could always dynamically generate it?

nsklyar replied:
the field is needed for the DB—to have a column with unique values in table `A2_ANNOTATIONSRC`. After moving biomart related fields in a separate table  table `ANNOTATIONSRC` didn't have any unique column (combination of columns) anymore, so `name` column was added.

olgamelnichuk commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/loader/biomart/BioMartAnnotator.java:
```diff
@@ -119,48 +106,45 @@ public void updateAnnotations(String annotationSrcId) {
             beDataWriter.writePropertyValues(data.getPropertyValues(), listener);
-            beDataWriter.writeBioEntityToPropertyValues(data, annSrc, listener);
+            beDataWriter.writeBioEntityToPropertyValues(data, annSrc, false, listener);
 
```
Can we guarantee here that all three `beDataWriter.*` calls be run in the same transaction?

nsklyar replied:
they are running in different transactions, but it a "feature" :) The steps shouldbe done in order, but they are independent.

olgamelnichuk commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/loader/biomart/BioMartConnection.java:
```diff
@@ -107,7 +119,9 @@ public String getOnlineMartVersion() throws BioMartAccessException {
-    public Collection<String> validateAttributeNames(Set<String> properties) throws BioMartAccessException {
+    @Override
+    public Collection<String> validateAttributeNames(Set<String> properties) throws AnnotationSourceAccessException {
+        connect();
```
Instead of checking `if (connected == "true")` in every method, it would be easier to create `BioMartConnection` with already pre-fetched registry info; just use `static` method instead of constructor: `BioMartConnection.create(bioMartSource)`...

nsklyar replied:
fixed

nsklyar committed:
create Annotators and Converters in AnnotationSrourceType

nsklyar committed:
removed unused method

olgamelnichuk commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/loader/AnnotationProcessor.java:
```diff
@@ -0,0 +1,80 @@
+
+    private Annotator getAnnotator(String id) {
+        final AnnotationSource annotationSource = fetchAnnSrcById(id);
+        final AnnotationSourceType annSrcType = AnnotationSourceType.getByClass(annotationSource.getClass());
```
why not to return the type by the annotationsource instance itself?
Probably bad idea.. 
or at least hide `annotationSource.getClass()`: e.g. `AnnotationSourceType.annTypeOf(annotationSource)`..

nsklyar replied:
makes sence, done

alf239 commented on annotator/src/main/java/uk/ac/ebi/gxa/annotator/model/AnnotationSource.java:
```diff
@@ -62,26 +63,68 @@
     }
 
+    @Column(name = "name")
+    public String getName() {
```
Having a getter with a side effect, _especially_ a side effect we rely on, is a very fragile idea: it's hard to spot, easy to forget, and super-easy to break later on. Why not create the name as we create a source?

nsklyar replied:
I've added call createName() into constructor, and remove from getter, but it's still easy to forget to call create name in new classes.

Commit:
Merge remote-tracking branch 'gxa/develop' into genesig

olgamelnichuk commented on annotator/src/test/java/uk/ac/ebi/gxa/annotator/dao/AnnotationSourceDAOTest.java:
```diff
@@ -93,17 +112,32 @@ public void testSave() throws Exception {
     @Test
     public void testGetById() {
+        AnnotationSource annotationSource = annSrcDAO.getById(1000, BioMartAnnotationSource.class);
+        Assert.assertNotNull(annotationSource);
```
Can we also be sure here that annotationSource is instance of BioMartAnnotationSource class?
I mean:

```
BioMartAnnotationSource annotationSource = annSrcDAO.getById(1000, BioMartAnnotationSource.class);
Assert.assertNotNull(annotationSource);
```

nsklyar replied:
data are taken fom our tast DB where annotation source with id 1000 is BioMartAnnotationSource

nsklyar committed:
renamed method getByClass to annSrcTypeOf

nsklyar committed:
changed package structure

nsklyar committed:
optimized impots

nsklyar committed:
Merge branch 'genesig' of github.com:gxa/gxa into genesig

nsklyar committed:
removed logic from getter

nsklyar committed:
created createConnection method to initialize connection

Commit:
Code formatting

Commit:
Code formatting

nsklyar committed:
removed obsolete annotation

nsklyar committed:
Merge branch 'genesig' of github.com:gxa/gxa into genesig

nsklyar committed:
fixes to address comments, use FileBasedAnnotationSource in Annotators

nsklyar committed:
added Copyright

nsklyar committed:
use StringBuilder

nsklyar committed:
fixes to address comments

nsklyar committed:
removing JUnit3 methods

nsklyar committed:
fixed test

nsklyar committed:
minor changes, addressed comments

Commit:
db unit tests clean up: no more  "extends DataSourceBasedDBTestCase"

Commit:
code simplification; weakened the access levels of default constructors used by hibernate..

Commit:
removed unused constructor

Commit:
Loader interface should have "load" method

Commit:
removing default constructor

Commit:
method isValidData() moved to data.isValid()

Commit:
a counter need to be incremented from-time-to-time

Commit:
Merge remote branch 'origin/develop' into genesig

Conflicts:
	atlas-web/src/main/java/uk/ac/ebi/gxa/tasks/AnnotationLoaderTask.java

Commit:
code cleanup: rename AnnotationLoader -> AnnotationCommandRunner

Commit:
code cleanup:
- fixed java doc
- AnnotationCommandListener methods were renamed

Commit:
code cleanup: eliminating probable NPE

Commit:
code cleanup: removed unused constructor

Commit:
extracted BioMartQuery class

olgamelnichuk committed:
typo

olgamelnichuk committed:
functionality of BETypeExternalAttributesHandler moved to AnnotationSource class

olgamelnichuk committed:
CSVBasedReader utility added; a wrapper over CSVReader to minimize code duplication

olgamelnichuk committed:
BioMartQuery renamed to MartQuery and moved to the biomart package;
BioMartQueryTest renamed to MartQueryTest and moved to the biomart package;
"count" field added to the MartQuery to specify if the query should return total number of results;

olgamelnichuk committed:
Added MartServiceClient;
Added apache httpclient 4.x to the annotator module dependencies;

olgamelnichuk committed:
added equals() and hashCode() methods to be able to compare organism objects by id and name;

olgamelnichuk committed:
added unit test for annotSource.getBioEntityType(name)

olgamelnichuk committed:
using httpclient 4.x now

olgamelnichuk committed:
splitting up annotation data loading classes to easy test and support:
- Extracted BioMartAnnotationLoader and GeneSigAnnotationLoader from Annotator code;
- Removed hierarchy from annotation data classes;
- Removed usage of java generics in Annotator classes;
- A big annotation parser splitted into pieces to easy read and test;
- BioMartDbDAO moved to biomart package

olgamelnichuk committed:
optimized imports

Commit:
removed unused code

olgamelnichuk committed:
making one instance of httpClient per application

Commit:
Merge remote branch 'origin/genesig2' into genesig2

Commit:
copyright added

Commit:
martServiceClient not implements Closeable anymore

nsklyar committed:
added methods

nsklyar committed:
Merge remote branch 'origin/genesig2' into genesig2

Commit:
Extracted interface from MartServiceClient

Commit:
unfinished and not compilable test code was committed by mistake

Commit:
fixing test after MartServiceClient interface extraction

Commit:
added @Override annotation

Commit:
MartBioEntitiesLoader test.. TODO

olgamelnichuk committed:
- "return" statement forgotten;
 - message in IllegalStateException forgotten;

olgamelnichuk committed:
BioEntitiesLoader unit test completed

olgamelnichuk committed:
- "return" statement forgotten

olgamelnichuk committed:
added MartPropertyValuesLoader unit test

Commit:
Added copyright

Commit:
optimized imports

Commit:
added a few lines of javadoc

Commit:
added MartDesignElementMappingsLoader unit test

Commit:
removed unused resource files and test

nsklyar committed:
use Validators instead of connection for validation

nsklyar committed:
removed unused method

nsklyar committed:
use Validators instead of connection for validation

nsklyar committed:
refactoring: dropped *Connections, introduced Validators, Managers

nsklyar committed:
rolled back accidental change

nsklyar committed:
added tests

nsklyar committed:
renamed method

nsklyar committed:
removed wrong import

nsklyar committed:
extract validation for single annotation source

nsklyar committed:
fixed admin page for Validate

nsklyar committed:
minor  fixes for generics

Commit:
copyright added

Commit:
removed countQuery from MartClient as it is useless

nsklyar committed:
added form validation

nsklyar committed:
Merge branch 'genesig' of github.com:gxa/gxa into genesig

Commit:
Merge remote branch 'origin/develop' into genesig

Conflicts:
	annotator/src/test/java/uk/ac/ebi/gxa/annotator/loader/annotationsrc/BioMartAnnotationSourceLoaderTest.java
	atlas-analytics/src/test/java/uk/ac/ebi/gxa/analytics/generator/TestExperimentAnalyticsGeneratorService.java
	atlas-dao/src/test/java/uk/ac/ebi/gxa/dao/AtlasDAOTestCase.java
	atlas-dao/src/test/resources/atlas-be-db.xml

Commit:
all column names converted to upper case as in the database

rpetry committed:
Merge branch 'develop' into genesig

nsklyar committed:
removed unused code

rpetry committed:
Merge branch 'develop' into genesig

Conflicts:
	atlas-dao/src/test/resources/atlas-be-db.xml

rpetry committed:
genesig - fixed junit tests

nsklyar committed:
fix: use external property names for genesigdb column names

nsklyar committed:
Merge branch 'genesig' of github.com:gxa/gxa into genesig

rpetry committed:
Merge branch 'develop' into genesig

rpetry committed:
genesig - Made annotations batch update size configurable via AtlasProperties.

rpetry committed:
Merge branch 'develop' into genesig
